### PR TITLE
Refactor/separate notebook dns module

### PR DESCRIPTION
**Pull request by caetano-colin**

### CODE REVIEW

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# ML DNS Notebooks
+
+Configures the service endpoints to use a custom Private Service Connect IP.
+
+- *.notebooks.cloud.google.com
+- *.notebooks.googleusercontent.com
+- *.kernels.googleusercontent.com
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/README.md]:
```suggestion
- `*.notebooks.cloud.google.com`
- `*.notebooks.googleusercontent.com`
- `*.kernels.googleusercontent.com`
```

File: 3-networks-dual-svpc/modules/base_env/main.tf [hunk 2]
```diff
@@ -234,6 +234,13 @@ module "restricted_shared_vpc" {
   perimeter_projects = [local.logging_env_project_number, local.kms_env_project_number]
 }
 
+module "ml_dns_vertex_ai" {
+  source                             = "../ml_dns_notebooks"
+  project_id                         = local.restricted_project_id
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/base_env/main.tf]:
```suggestion
  source = "../ml_dns_notebooks"

  project_id                         = local.restricted_project_id
```

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# ML DNS Notebooks
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/README.md]:
```suggestion
# Machine Learning Cloud DNS Notebooks Service Endpoints module
```

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+/***********************************************
+  Notebooks DNS Zone & records.
+ ***********************************************/
+
+module "notebooks" {
+  source      = "terraform-google-modules/cloud-dns/google"
+  version     = "~> 5.0"
+  project_id  = var.project_id
+  type        = "private"
+  name        = "dz-${local.project_suffix_env_code}-shared-restricted-notebooks"
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf]: this should be a variable without default

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  cidr_prefix = split("/", local.cidr_block)[1]
+
+  # Calculate the number of available IP addresses
+  ip_count = range(pow(2, 32 - local.cidr_prefix))
+
+  # Generate a list of IP addresses
+  google_private_ip_addresses = [for i in range(pow(2, 32 - local.cidr_prefix)) : cidrhost(local.cidr_block, i)]
+
+  project_labels          = data.google_project.project.labels
+  project_suffix_env_code = contains(keys(local.project_labels), "env_code") ? local.project_labels.env_code : ""
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf]: replace with direct variables for the names

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+ * limitations under the License.
+ */
+
+data "google_netblock_ip_ranges" "private_apis" {
+  range_type = "private-googleapis"
+}
+
+data "google_project" "project" {
+  project_id = var.project_id
+}
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf]: replace with direct variables for the names

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    },
+  ]
+}
+
+module "notebooks-googleusercontent" {
+  source      = "terraform-google-modules/cloud-dns/google"
+  version     = "~> 5.0"
+  project_id  = var.project_id
+  type        = "private"
+  name        = "dz-${local.project_suffix_env_code}-shared-restricted-notebooks-googleusercontent"
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf]: this should be a variable without default

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    },
+  ]
+}
+
+module "kernels-googleusercontent" {
+  source      = "terraform-google-modules/cloud-dns/google"
+  version     = "~> 5.0"
+  project_id  = var.project_id
+  type        = "private"
+  name        = "dz-${local.project_suffix_env_code}-shared-restricted-kernels-googleusercontent"
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf]: this should be a variable without default

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/variables.tf [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  type        = string
+  description = "Project ID of the project in which Cloud DNS configurations will be made."
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/variables.tf]: be consistent with the ordering of the attributes
```suggestion
  description = "Project ID of the project in which Cloud DNS configurations will be made."
  type        = string
```

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/variables.tf [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+variable "private_service_connect_ip" {
+  type        = string
+  description = "Google Private Service Connect IP Address."
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/variables.tf]:
```suggestion
  description = "Google Private Service Connect IP Address."
  type        = string
```

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/variables.tf [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+variable "private_visibility_config_networks" {
+  description = "List of VPC self links that can see this zone."
+  default     = []
+  type        = list(string)
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/variables.tf]:
```suggestion
  type        = list(string)
  default     = []
```

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+      ttl     = 300
+      records = [var.private_service_connect_ip]
+    },
+  ]
+}
+
+module "kernels-googleusercontent" {
+  source      = "terraform-google-modules/cloud-dns/google"
+  version     = "~> 5.0"
+  project_id  = var.project_id
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf]:
```suggestion
  source  = "terraform-google-modules/cloud-dns/google"
  version = "~> 5.0"

  project_id  = var.project_id
```

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+      ttl     = 300
+      records = [var.private_service_connect_ip]
+    },
+  ]
+}
+
+module "notebooks-googleusercontent" {
+  source      = "terraform-google-modules/cloud-dns/google"
+  version     = "~> 5.0"
+  project_id  = var.project_id
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf]:
```suggestion
  source  = "terraform-google-modules/cloud-dns/google"
  version = "~> 5.0"

  project_id  = var.project_id
```

File: 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+/***********************************************
+  Notebooks DNS Zone & records.
+ ***********************************************/
+
+module "notebooks" {
+  source      = "terraform-google-modules/cloud-dns/google"
+  version     = "~> 5.0"
+  project_id  = var.project_id
```
Comment (daniel-cit) [inline 3-networks-dual-svpc/modules/ml_dns_notebooks/main.tf]:
```suggestion
  source  = "terraform-google-modules/cloud-dns/google"
  version = "~> 5.0"

  project_id  = var.project_id
```

**Review by daniel-cit — CHANGES_REQUESTED** [review]
- Formatting fixes
- Replacing `data` with variable

**Comment by caetano-colin** [comment]
@daniel-cit implemented the changes, back to you for review

**Pull request closed without merging**